### PR TITLE
feat: prepare for TS defs generation

### LINKED ISSUE #53: Generate TypeScript definitions for P3
Child of vaadin/vaadin-core#186

### PR DESCRIPTION
**Pull request by Haprog**
Fixes #53 

Generated file `vaadin-item.d.ts` looks like this:
```ts
import {PolymerElement} from '@polymer/polymer/polymer-element.js';

import {ThemableMixin} from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';

import {DirMixin} from '@vaadin/vaadin-element-mixin/vaadin-dir-mixin.js';

import {ItemMixin} from './vaadin-item-mixin.js';

import {html} from '@polymer/polymer/lib/utils/html-tag.js';

declare class ItemElement extends
  ItemMixin(
  ThemableMixin(
  DirMixin(
  PolymerElement))) {
  value: string;
}

declare global {
  interface HTMLElementTagNameMap {
    "vaadin-item": ItemElement;
  }
}

export {ItemElement};
```

Generated file `vaadin-item-mixin.d.ts` looks like this:
```ts
export {ItemMixin};

declare function ItemMixin<T extends new (...args: any[]) => {}>(base: T): T & ItemMixinConstructor;

interface ItemMixinConstructor {
  new(...args: any[]): ItemMixin;
}

export {ItemMixinConstructor};

interface ItemMixin {
  value: string;
  _hasVaadinItemMixin: boolean;
  disabled: boolean;
  selected: boolean;
  ready(): void;
  disconnectedCallback(): void;
  _setFocused(focused: boolean): void;
  _setActive(active: boolean): void;
  _onKeydown(event: KeyboardEvent): void;
  _onKeyup(event: KeyboardEvent): void;
}
```

### CODE REVIEW

File: src/vaadin-item-mixin.html [hunk 1]
```diff
@@ -133,13 +152,15 @@
...
 
+    /** @private */
     _onKeydown(event) {
       if (/^( |SpaceBar|Enter)$/.test(event.key) && !event.defaultPrevented) {
         event.preventDefault();
         this._setActive(true);
       }
     }
 
+    /** @private */
```
Comment (web-padawan) [inline src/vaadin-item-mixin.html:163]: Please mark this listener as `@protected`, we [override it](https://github.com/vaadin/vaadin-tabs/blob/84cd9c794b113a6f429c717f0687bb381e98eb8d/src/vaadin-tab.html#L59-L62) in `vaadin-tab`.

File: src/vaadin-item-mixin.html [hunk 2]
```diff
@@ -133,13 +152,15 @@
...
```
Comment (web-padawan) [inline src/vaadin-item-mixin.html:155]: Can be also marked as `@protected` (listener isn't bound and can be reused).
Not a blocker though.

File: src/vaadin-item-mixin.html (hunk 1 above)
Comment (Haprog) [reply src/vaadin-item-mixin.html]: Done.

File: src/vaadin-item-mixin.html (hunk 2 above)
Comment (Haprog) [reply src/vaadin-item-mixin.html]: Done.

**Comment by Haprog** [comment]
> Please add @mixes Vaadin.DirMixin to vaadin-item JSDoc annotation.

Done.

**Pull request merged**